Detectors/GlobalTracking/include/GlobalTracking/TrackCuts.h

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file TrackCuts.h
+/// \brief Class to perform track cuts
+/// \author amelia.lindner@cern.ch
+
+#ifndef ALICEO2_TRACKCUTS_H
+#define ALICEO2_TRACKCUTS_H
+
+#include "Framework/Logger.h"
+#include "Framework/DataTypes.h"
+#include "ReconstructionDataFormats/Track.h"
+#include "DataFormatsGlobalTracking/RecoContainer.h"
+#include "ReconstructionDataFormats/MatchInfoTOF.h"
+#include "DataFormatsTPC/TrackTPC.h"
+#include "GlobalTracking/TrackMethods.h"
+#include "DetectorsBase/Propagator.h"
+#include <set>
+#include <vector>
+#include "Rtypes.h"
+#include <gsl/span>
+namespace o2
+{
+using GID = o2::dataformats::GlobalTrackID;
+class TrackCuts
+{
+ public:
+  //////////////////////////////// O2 ////////////////////////////////
+  bool isSelected(GID trackIndex, o2::globaltracking::RecoContainer& data)
+  {
+    o2::track::TrackParCov trk;
+    auto contributorsGID = data.getSingleDetectorRefs(trackIndex);
+    auto src = trackIndex.getSource(); // make selections depending on track source
+    if (src == GID::ITSTPCTRDTOF) {
+      trk = data.getTrack<o2::track::TrackParCov>(data.getITSTPCTRDTOFMatches()[trackIndex].getTrackRef()); // ITSTPCTRDTOF is ITSTPCTRD + TOF cluster
+    } else if (src == GID::TPCTRDTOF) {
+      trk = data.getTrack<o2::track::TrackParCov>(data.getTPCTRDTOFMatches()[trackIndex].getTrackRef()); // TPCTRDTOF is TPCTRD + TOF cluster
+    } else if (src == GID::ITSTPCTOF) {
+      trk = data.getTrack<o2::track::TrackParCov>(data.getTOFMatch(trackIndex).getTrackRef()); // ITSTPCTOF is ITSTPC + TOF cluster
+    } else if (src == GID::TPC) {                                                              // TPC tracks selection
+      const auto& tpcTrk = data.getTPCTrack(contributorsGID[GID::TPC]);
+      uint8_t tpcNClsShared, tpcNClsFound, tpcNClsCrossed, tpcNClsFindable, tpcChi2NCl;
+      // tpcNClsFindable = tpcTrk.getNClusters();
+      // tpcChi2NCl = tpcTrk.getNClusters() ? tpcTrk.getChi2() / tpcTrk.getNClusters() : 0;
+      // o2::TrackMethods::countTPCClusters(tpcTrk, data.getTPCTracksClusterRefs(), data.clusterShMapTPC, data.getTPCClusters(), tpcNClsShared, tpcNClsFound, tpcNClsCrossed);
+      // double tpcCrossedRowsOverFindableCls = tpcNClsCrossed / tpcNClsFindable;
+      math_utils::Point3D<float> v{};
+      std::array<float, 2> dca;
+      if (tpcTrk.getPt() < mPtTPCCut ||
```

Comment:
is the aim to apply TPC-track specific cut? If so, all track types above do contain TPC track, they also should be processed in the same way. See how the track contributors are extracted from all types of barrel tracks: https://github.com/AliceO2Group/AliceO2/blob/f87c7843b23994456f25d0ccff4dc5c01ccf4bf4/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L1725-L1793

Reply:
i changedhow the track source is found in the same way as it is done in AODProducerWorkflowSpec.cxx. Can you please check?